src/providers/authenticationProvider.ts

```diff
@@ -104,15 +107,65 @@ export class TerraformCloudAuthenticationProvider implements vscode.Authenticati
   async createSession(_scopes: readonly string[]): Promise<vscode.AuthenticationSession> {
     this.ensureInitialized();
 
-    // Prompt for the UAT.
-    const token = await vscode.window.showInputBox({
-      ignoreFocusOut: true,
-      placeHolder: 'User access token',
-      prompt: 'Enter an HashiCorp Terraform User Access Token (UAT).',
-      password: true,
-    });
+    vscode.window.showInformationMessage;
+    const choice = await vscode.window.showQuickPick(
+      [
+        {
+          label: 'Stored User Token',
+          detail: 'Use a token stored in the Terraform CLI configuration file',
+        },
+        {
+          label: 'Existing User Token',
+          detail: 'Enter a token manually',
+        },
+        {
+          label: 'Open to generate a User token',
+          detail: 'Open the Terraform Cloud website to generate a new token',
+        },
+      ],
+      {
+        canPickMany: false,
+        ignoreFocusOut: true,
+        placeHolder: 'Choose a method to enter a Terraform Cloud User Token',
+        title: 'HashiCorp Terraform Cloud Authentication',
+      },
+    );
+    if (choice === undefined) {
+      throw new Error('Must login to continue');
+    }
 
-    // Note: this example doesn't do any validation of the token beyond making sure it's not empty.
+    // TODO: Change to production URL
+    const terraformCloudURL = 'https://app.staging.terraform.io/app/settings/tokens?source=terraform-login';
+
+    let token: string | undefined = '';
+    switch (choice.label) {
+      case 'Open Terraform Cloud to generate a token':
+        await vscode.env.openExternal(vscode.Uri.parse(terraformCloudURL));
+        // Prompt for the UAT.
+        token = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: 'User access token',
+          prompt: 'Enter an HashiCorp Terraform User Access Token (UAT).',
+          password: true,
+        });
+        break;
+      case 'Enter a existing Terraform Cloud User Token':
```

Comment:
```suggestion
      case 'Existing User Token':
```
nitpick: might want to extract that into a constant to prevent similar mistakes in the future